test/e2e/apps/deployment.go

```diff
@@ -388,6 +388,8 @@ var _ = SIGDescribe("Deployment", func() {
 		framework.ExpectNoError(err, "failed to see replicas of %v in namespace %v scale to requested amount of %v", testDeployment.Name, testNamespaceName, testDeploymentDefaultReplicas)
 
 		ginkgo.By("patching the DeploymentStatus")
+		// fetch new deploymentsList.ResourceVersion before patching the DeploymentStatus
+		deploymentsList, err = f.ClientSet.AppsV1().Deployments("").List(context.TODO(), metav1.ListOptions{LabelSelector: testDeploymentLabelsFlat})
```

Comment:
Isn't it necessary to put an error handling for `err` with `framework.ExpectNoError(err)`?

Reply:
Just because we are sending a patch, it is not needed to fetch the deployment list. Other than that, you're trying to fetch a list of deployments from `""` namespace, which is also not correct.

Reply:
@soltysh without this 

https://github.com/sayaoailun/kubernetes/blob/8a3303b5367e60171513872cb683c89038f6ce52/test/e2e/apps/deployment.go#L399

we use `deploymentsList` from here

https://github.com/sayaoailun/kubernetes/blob/8a3303b5367e60171513872cb683c89038f6ce52/test/e2e/apps/deployment.go#L319

the event here

https://github.com/sayaoailun/kubernetes/blob/8a3303b5367e60171513872cb683c89038f6ce52/test/e2e/apps/deployment.go#L419-L424

 we got is this

https://github.com/sayaoailun/kubernetes/blob/8a3303b5367e60171513872cb683c89038f6ce52/test/e2e/apps/deployment.go#L347

not this

https://github.com/sayaoailun/kubernetes/blob/8a3303b5367e60171513872cb683c89038f6ce52/test/e2e/apps/deployment.go#L412

Reply:
The difference between your fix and what we currently have is that we'll start the watch in https://github.com/kubernetes/kubernetes/blob/a27498ef42f27e7cddadb2c5b31bc80fed81d964/test/e2e/apps/deployment.go#L414 with a newer RV, but that's not required, and most importantly still doesn't fix the problem in this code. I'm working on an actual fix, but that will require a bigger change in this code.

Reply:
https://github.com/kubernetes/kubernetes/pull/129110 will correctly handle this situation.